File: medolylibrary/src/main/java/com/wa2c/android/medoly/library/IProperty.java
package com.wa2c.android.medoly.library;


import android.content.Context;
import android.support.annotation.NonNull;

/**
 * Property interface.
 */
public interface IProperty {
    /** Get property name id. */
    int getNameId();
    /** Get the property name. */
    String getName(@NonNull Context context);
    /** Get the property key name. */
    String getKeyName();
}

File: medolylibrary/src/main/java/com/wa2c/android/medoly/library/IProperty.kt
package com.wa2c.android.medoly.library


import android.content.Context
import android.os.Parcelable

/**
 * Property interface.
 */
interface IProperty : Parcelable {
    /** Get property name id.  */
    val nameId: Int
    /** Get the property key name.  */
    val keyName: String

    /** Get the property name.  */
    fun getName(context: Context): String
}
